### ISSUE TITLE
Factions vs FactionsUUID?

### ISSUE THREAD

**Issue by HunterGPlays** [MassiveCraft/Factions#917]
Right now, I own FactionsUUID and its a bit of a pain considering that its v1.6.x and it doesn't have the modern features like /f recruit, /f perm, /f truces etc. 

However I use FactionsUUID for:
- The faction finder (boat glitch) patch
- Faction warps
- Limiting allies and other relations
- /f stuck (if you stay within 10 blocks for 30 seconds, it teleports you to a safe location - useful when stuck in a trap)

Is there any way you could implement these features? The last version of factions I used was 2.7.5 so excuse me if I'm a bit mistaken on what Factions 2.8 already has.

**Comment by MarkehMe** [comment]
Hi! 

Can you explain the boat glitch please? I didn't think this was existent in the latest Factions, if it is i'll fix it personally now. 

Faction warps are something I made in FactionsPlus. I could easily release a FactionWarps plugin if @oloflarsson doesn't think warps should be in the core of Factions. 

Can you elaborate "Limiting allies and other relations"? 

/f stuck seems like a feature that shouldn't be in the core of Factions but could easily be made into an extension.

**Comment by HunterGPlays** [comment]
I believe the boat glitch is patched, however I'm not 100% sure. Basically, you enable the faction finder exploit, get in a boat, turn f map on, and enable the exploit. It spams your chat with tons of f map from all over the world.

By "Limiting allies and other relations", I meant imposing a limit on how many allies/truces/enemies you could have. For example, I could allow factions to only have 2 allies and 5 truces.

And to be honest, I only got FactionsUUID for the features.

**Comment by MarkehMe** [comment]
As FactionsUUID is Actually just Factions 1.6 with a few patches on-top of it it doesn't surprise me (1.6 is so old!) 

I don't think there is currently a way to limit the amount of allies/truces/enemies you can set but this is a simple addition, however if MassiveCraft doesn't want it as a feature I could again put that into an extension plugin! Same with warps.

Are they the only two major features you're missing?

**Comment by HunterGPlays** [comment]
Yes, I would be grateful if you could add these. I'm sure MassiveCraft wouldn't mind a toggleable config option for it. With faction warps, they should have a max limit and a price to set them.

**Comment by HunterGPlays** [comment]
Have you got an ETA on this @MarkehMe?

**Comment by MarkehMe** [comment]
@HunterGPlays I'm working on an extension plugin for it. Will put it up (for free) on Bukkit Plugins and Spigot Resources

**Comment by MarkehMe** [comment]
@HunterGPlays we've decided to add these to the main plugin:
* /f stuck - if you stay within x (configurable, default = 10) blocks for x (configureable, default = 30) seconds in x (configureable, default = Enemy, Ally) territories it will attempt to teleport you to the nearest chunk in a radius of x (default = 2) chunks
* Warps
* Limitations on enemies, allies, and truces 

Will reference this issue ticket as I progress. Will start on /f stuck first!

**Comment by HunterGPlays** [comment]
That sounds great!

**Comment by oloflarsson** [comment]
I gave `/f unstuck` a go: https://github.com/MassiveCraft/Factions/commit/3f3ee4df6d66a4ef493314ecff156498a4b543de

**Comment by HunterGPlays** [comment]
How would you guys feel about adding faction warps?

**Comment by ulumulu1510** [comment]
It is on our list of To-Do's, but besides @MarkehMe no one of our team is actively developing on Factions at the moment. Would you feel like implementing this MarkehMe? :)

**Comment by MarkehMe** [comment]
Sure, we can implement this into the main branch of Factions. I've been working on warps in one of my other projects - when it's feature-full I'll bring it over!

**Comment by ulumulu1510** [comment]
Great :) When ever you feel like it we can work on it together and see how it goes.
